hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/streaming/BlockState.java

```diff
@@ -38,36 +160,233 @@ public long blockNumber() {
     }
 
     /**
-     * Get the list of item bytes
+     * Add an item to the BlockState, this will not create a PublishStreamRequest.
+     * @param item the item to add
+     */
+    public void addItem(final BlockItem item) {
+        if (item == null) {
+            return;
+        }
+
+        if (closedTimestamp.get() != null) {
+            throw new IllegalStateException("Block is closed; adding more items is not permitted");
+        }
+
+        if (item.hasBlockHeader() && !headerItemInfo.addedInBlockState()) {
+            logger.warn(
+                    "[Block {}] Block header item added, but block header already encountered (state={})",
+                    blockNumber,
+                    headerItemInfo.state.get());
+        } else if (item.hasBlockProof() && !proofItemInfo.addedInBlockState()) {
```

Comment:
This condition should not have a side effect of setting on `state`. It is confusing to see this throw warning
```
item.hasBlockProof() && !proofItemInfo.addedInBlockState()
```